File: Leetcode_TS/dota2-senate.ts
function predictPartyVictoryy(senate: string): string {
    

    let radiant:number[] = []
    let dire: number[] = []

    for (let i = 0; i < senate.length; i++) {
        if (senate[i] === "R") radiant.push(i)
        if (senate[i] === "D") dire.push(i)
    }

    while (radiant.length && dire.length) {
        let r = radiant.shift() as number
        let d = dire.shift() as number

        if (r < d) radiant.push(r+senate.length)
        else dire.push(d+senate.length)
    }

   return radiant.length ? "Radiant": "Dire"
};

File: Leetcode_Python/dota2-senate.py
class Solution(object):
    def predictPartyVictory(self, senate):
        """
        :type senate: str
        :rtype: str
        """
        
        radiant = []
        dire = []

        for i in range(len(senate)):
            if senate[i] == "R":
                radiant.append(i)
            else:
                 dire.append(i)

        while len(radiant) > 0 and len(dire) > 0:
            rad = radiant.pop(0)
            d = dire.pop(0)

            if rad < d:
                radiant.append(rad + len(senate))
            else:
                dire.append(d + len(senate))

        if len(radiant) > 0:
            return "Radiant" 
        else:
            return "Dire"